backend/src/main/java/server/RhymeEngine.java

```diff
@@ -0,0 +1,88 @@
+package com.animalrhymes.server;
+
+import java.util.*;
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+/*
+ * The Rhyme Engine that gets a phonetic and matches it with a list of animals an returns that list
+ * in the utility method getAnimalsThatRhymeWith(String word)
+ */
+public class RhymeEngine {
+    private final static String FILENAME = "AnimalsAndPhonetic.csv";
+
+    private final static Map<String, List<String>> HM = new HashMap<>();
+    private final static boolean createdHashMap = false;
+    private final static String[] DEFAULT_EMPTY_LIST = new String[] {};
+
+    public final static RhymeEngine init() {
+        return new RhymeEngine();
+    }
+
+    // Create HashMap from phonetic –> list of animals
+    private RhymeEngine() {
+        if (!Files.exists(Paths.get(FILENAME))) {
+        	System.out.println("Creating file because does not exist...");
+            AnimalAndPhoneticFileCreator.createFile();
+			System.out.println("File created.");
+		}
+
+        if (!createdHashMap) {
+        	System.out.println("Creating HashMap for RhymeEngine...");
```

Comment:
Should createdHashMap be set to true at around line 62?

Reply:
Yep, nice catch I forgot to add that.